src/client/client.ts

```diff
@@ -92,11 +92,21 @@ class ClientRequestImpl {
     setBody = !(methodUpperCase === 'GET' || methodUpperCase === 'HEAD')
 
     // Pass URL string to 1st arg for testing with MSW and node-fetch
-    return (opt?.fetch || fetch)(url, {
+    const res = (opt?.fetch || fetch)(url, {
       body: setBody ? this.rBody : undefined,
       method: methodUpperCase,
       headers: headers,
     })
+
+    if (opt?.after) {
+      const hookedResponse = await opt.after(await res)
+      if (hookedResponse) {
+        return hookedResponse
+      } else {
+        return res
+      }
```

Comment:
This could be `return hookedResponse ?? res`, couldn't it? Line 101&ndash;110 could actually be reduced to `return (opt?.after && await opt.after(await res)) ?? res`, but it's a bit too extreme. 😅

Reply:
Ah, yes! `return hookedResponse ?? res` is good, but `return (opt?.after && await opt.after(await res)) ?? res` is too  extreme :)

Reply:
Glad you liked the first suggestion! I noticed I can be even more extreme with `return opt?.after?.(await res) ?? res` lol.

Reply:
Haha